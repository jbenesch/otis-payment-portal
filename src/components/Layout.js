import React from 'react'
import PropTypes from 'prop-types'
import { Page, Row, Column } from 'hedron'
import { Link } from 'redux-little-router'
import Transition from 'react-transition-group/Transition'
import styled from 'styled-components'
import { Colors } from '@blueprintjs/core'
import AccountMenu from './AccountMenu'
import Sidebar from './Sidebar'
import { Tablet } from '../components/BreakPoint'
import Logo from '../assets/logo.svg'

const Title = styled.h1`
  color: ${Colors.LIGHT_GRAY5};
  font-weight: normal;
  margin: 0 0 5px 15px;
  font-size: 2.2em;
  font-family: inherit;
`

const SubTitle = styled.h2`
  color: ${Colors.LIGHT_GRAY2};
  font-weight: normal;
  margin: 0 0 0 15px;
  font-size: 1.2em;
`

const Viewport = styled.div`
  background: ${Colors.DARK_GRAY1};
  border-radius: 24px;
  opacity: 0.98;
  width: 100%;
`

const defaultStyle = {
  transition: `all 500ms ease-in-out`,
  transform: 'translateY(-20px)',
  opacity: 0
}

const transitionStyles = {
  entering: {
    opacity: 0,
    transform: 'translateY(-20px)'
  },
  entered: {
    opacity: 1,
    transform: 'translateY(0px)'
  }
}

const Layout = ({ children, title, subtitle }) => (
  <Page fluid className="pt-dark">
    <Row alignItems="center" justifyContent="space-between">
      <Column sm={1.5} style={{ padding: '10px 20px', textAlign: 'center' }}>
        <Link href="/">
          <img
            src={Logo}
            alt="Otis Payments"
            style={{ width: 70, marginTop: 5 }}
          />
        </Link>
      </Column>
      <Column sm={7.5} style={{ padding: '10px 20px' }}>
        <Title>{title}</Title>
        <SubTitle>{subtitle}</SubTitle>
      </Column>
      <Column sm={3} style={{ padding: '10px 20px' }}>
        <AccountMenu />
      </Column>
    </Row>
    <Row>
      <Tablet larger>
        <Column sm={1.5} style={{ padding: '0 0 0 10px' }}>
          <Sidebar />
        </Column>
        <Column sm={10.5} style={{ padding: '0px 20px 0 0' }}>
          <Row>
            <Column lg={12} style={{ padding: '0px 10px' }}>
              <Viewport>
                <Transition in timeout={10} appear>
                  {state => (
                    <div
                      style={{
                        ...defaultStyle,
                        ...transitionStyles[state]
                      }}
                    >
                      {children}
                    </div>
                  )}
                </Transition>
              </Viewport>
            </Column>
          </Row>
        </Column>
      </Tablet>
      <Tablet smaller>
        <Column sm={12} style={{ padding: '0px 20px' }}>
          <Row>
            <Column lg={12} style={{ padding: '0px 10px' }}>
              <Viewport>
                <Transition in timeout={10} appear>
                  {state => (
                    <div
                      style={{
                        ...defaultStyle,
                        ...transitionStyles[state]
                      }}
                    >
                      {children}
                    </div>
                  )}
                </Transition>
              </Viewport>
            </Column>
          </Row>
        </Column>
      </Tablet>
    </Row>
  </Page>
)

Layout.propTypes = {
  children: PropTypes.element.isRequired,
  title: PropTypes.string,
  subtitle: PropTypes.string
}

Layout.defaultProps = {
  title: 'My Dashboard',
  subtitle: 'Welcome to Otis payment portal'
}

export default Layout
