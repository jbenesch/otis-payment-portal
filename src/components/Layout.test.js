import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import Layout from './Layout'

const mockStore = configureStore([])
const store = mockStore({
  router: {
    pathname: '/',
    route: '/',
    result: {
      id: 'Dashboard'
    }
  }
})

it('matches a previous snapshot', () => {
  window.matchMedia = jest.fn(() => ({
    matches: true,
    addListener: jest.fn()
  }))
  const component = renderer.create(
    <Provider store={store}>
      <Layout>
        <h1>Hello World</h1>
      </Layout>
    </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render the layout', () => {
  window.matchMedia = jest.fn(() => ({
    matches: true,
    addListener: jest.fn()
  }))
  const wrapper = shallow(
    <Provider store={store}>
      <Layout>
        <h1>Hello World</h1>
      </Layout>
    </Provider>
  )
  expect(wrapper).toMatchSnapshot()
})

it('should contain hello world in viewport', () => {
  window.matchMedia = jest.fn(() => ({
    matches: true,
    addListener: jest.fn()
  }))
  const wrapper = shallow(
    <Provider store={store}>
      <Layout>
        <h1>Hello World</h1>
      </Layout>
    </Provider>
  )
  expect(wrapper.contains(<h1>Hello World</h1>)).toEqual(true)
})
