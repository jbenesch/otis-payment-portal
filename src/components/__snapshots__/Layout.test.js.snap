// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a previous snapshot 1`] = `
<div
  className="pt-dark sc-bxivhb jcNPuI"
>
  <section
    className="sc-bwzfXH gSjTjY"
  >
    <div
      className="sc-htpNat eGJZKr"
      style={
        Object {
          "padding": "10px 20px",
          "textAlign": "center",
        }
      }
    >
      <a
        href="/"
        onClick={[Function]}
        target={undefined}
      >
        <img
          alt="Otis Payments"
          src="logo.svg"
          style={
            Object {
              "marginTop": 5,
              "width": 70,
            }
          }
        />
      </a>
    </div>
    <div
      className="sc-htpNat lgdQgR"
      style={
        Object {
          "padding": "10px 20px",
        }
      }
    >
      <h1
        className="sc-gzVnrw crYLJH"
      >
        My Dashboard
      </h1>
      <h2
        className="sc-htoDjs gMVxtF"
      >
        Welcome to Otis payment portal
      </h2>
    </div>
    <div
      className="sc-htpNat cpDYXL"
      style={
        Object {
          "padding": "10px 20px",
        }
      }
    >
      <span
        className="pt-popover-wrapper"
      >
        <div
          className="pt-popover-target"
          onClick={[Function]}
        >
          <div
            className="sc-EHOje hQOsTd"
            disabled={undefined}
            tabIndex={undefined}
          >
            <img
              className="sc-ifAKCX gZppUQ"
              size="medium"
              src=""
              style={
                Object {
                  "marginRight": 15,
                }
              }
            />
            <span>
              Hi 
              <b>
                
              </b>
               
              <svg
                className="pt-icon"
                data-icon="chevron-down"
                height={20}
                style={Object {}}
                viewBox="0 0 20 20"
                width={20}
              >
                <title>
                  chevron-down
                </title>
                <path
                  d="M16 6c-.28 0-.53.11-.71.29L10 11.59l-5.29-5.3a1.003 1.003 0 0 0-1.42 1.42l6 6c.18.18.43.29.71.29s.53-.11.71-.29l6-6A1.003 1.003 0 0 0 16 6z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
      </span>
    </div>
  </section>
  <section
    className="sc-bwzfXH irrVft"
  >
    <div
      className="sc-htpNat eGJZKr"
      style={
        Object {
          "padding": "0 0 0 10px",
        }
      }
    >
      <ul
        style={
          Object {
            "marginTop": 0,
            "paddingLeft": 0,
          }
        }
      >
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            className="active"
            href="/"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="dashboard"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                dashboard
              </title>
              <path
                d="M6 5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zM4 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm6-4c.55 0 1-.45 1-1s-.45-1-1-1-1 .45-1 1 .45 1 1 1zm0-5C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm6-9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm-8 5c0 1.1.9 2 2 2s2-.9 2-2c0-.33-2-8-2-8s-2 7.67-2 8zm6-9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                fillRule="evenodd"
              />
            </svg>
             
            My Dashboard
          </a>
        </li>
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            href="/account"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="bank-account"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                bank-account
              </title>
              <path
                d="M19.2 8.02l-.78-.18C18.03 6.4 17.2 5.08 16.08 4l.5-2.28c.11-.42-.22-.78-.61-.72-1.06.12-2 .54-2.67 1.26-1.06-.42-2.34-.66-3.56-.66-3.12 0-5.79 1.5-7.4 3.72-.23-.05-.45-.11-.67-.11C.72 5.21 0 5.98 0 7c0 .72.39 1.32.95 1.62-.06.42-.12.9-.12 1.38 0 2.16.89 4.08 2.28 5.58l-.33 2.04c-.11.72.45 1.38 1.12 1.38h.72c.56 0 1-.42 1.11-1.02l.06-.48c1.17.54 2.5.9 3.95.9 1.39 0 2.78-.3 3.95-.9l.06.48c.11.6.56 1.02 1.11 1.02h.72c.67 0 1.22-.66 1.11-1.38l-.33-1.98c.78-.78 1.34-1.74 1.73-2.76l1-.24c.5-.12.89-.6.89-1.2V9.22c.11-.6-.28-1.08-.78-1.2zM15 10c-.6 0-1-.7-1-1.5S14.4 7 15 7s1 .7 1 1.5-.4 1.5-1 1.5zM7.55 5.83a.99.99 0 0 1-1.38-.28.99.99 0 0 1 .28-1.38c2.34-1.56 4.77-1.56 7.11 0 .46.31.58.93.28 1.39-.31.46-.93.58-1.39.28-1.67-1.12-3.23-1.12-4.9-.01z"
                fillRule="evenodd"
              />
            </svg>
             
            Account
          </a>
        </li>
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            href="/mobile"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="mobile-phone"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                mobile-phone
              </title>
              <path
                d="M15 0H5c-.55 0-1 .45-1 1v18c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1zm-5 19c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm4-3H6V3h8v13z"
                fillRule="evenodd"
              />
            </svg>
             
            Mobile
          </a>
        </li>
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            href="/bills"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="dollar"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                dollar
              </title>
              <path
                d="M15.57 11.19c-.27-.51-.63-.93-1.07-1.26-.44-.33-.95-.6-1.51-.79-.56-.2-1.14-.36-1.72-.5-.6-.14-1.19-.26-1.75-.38-.57-.13-1.07-.27-1.51-.44-.44-.17-.8-.38-1.07-.63s-.41-.59-.41-1c0-.33.09-.6.28-.81.19-.21.42-.36.69-.47.27-.11.57-.18.88-.22.31-.04.58-.06.8-.06.71 0 1.35.14 1.9.41.55.27.91.81 1.06 1.62h3.36c-.09-.84-.32-1.56-.69-2.16-.37-.6-.83-1.08-1.38-1.45-.56-.37-1.18-.64-1.86-.81-.19-.05-.38-.07-.57-.1V1c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1v1.1c-.22.03-.43.05-.66.1-.73.13-1.39.37-1.98.71-.6.34-1.09.8-1.47 1.35-.39.56-.58 1.25-.58 2.08 0 .76.13 1.41.4 1.93.26.52.62.95 1.06 1.28.44.33.94.6 1.5.79.55.2 1.13.36 1.74.5.58.14 1.16.26 1.72.38s1.07.26 1.51.43c.44.17.8.39 1.09.66.28.27.43.63.45 1.06.02.43-.08.78-.3 1.04-.22.26-.49.47-.83.6-.34.14-.7.23-1.09.28-.39.05-.73.07-1.03.07-.87 0-1.61-.2-2.23-.59-.62-.39-.98-1.08-1.07-2.06H3c.02.9.19 1.68.52 2.34.33.66.78 1.21 1.35 1.65.57.44 1.25.77 2.03.98.35.1.71.16 1.08.21V19c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.13c.25-.04.5-.07.76-.13.77-.18 1.47-.46 2.1-.85.63-.39 1.14-.9 1.54-1.53.4-.63.59-1.39.59-2.29.01-.75-.13-1.37-.4-1.88z"
                fillRule="evenodd"
              />
            </svg>
             
            Bills
          </a>
        </li>
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            href="/complaints"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="chat"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                chat
              </title>
              <path
                d="M19 0H7c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h5.59l3.71 3.71c.17.18.42.29.7.29.55 0 1-.45 1-1v-3h1c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1zM7 13c-1.1 0-2-.9-2-2V4H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h1v3a1.003 1.003 0 0 0 1.71.71L7.41 16H13c.55 0 1-.45 1-1v-.17L12.17 13H7z"
                fillRule="evenodd"
              />
            </svg>
             
            Complaints
          </a>
        </li>
        <li
          className="sc-bZQynM ddVHqk"
        >
          <a
            href="/support"
            onClick={[Function]}
            target={undefined}
          >
            <svg
              className="pt-icon"
              data-icon="phone"
              height={20}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              viewBox="0 0 20 20"
              width={20}
            >
              <title>
                phone
              </title>
              <path
                d="M19.91 15.51c-.08-.08-4.21-2.5-4.35-2.57a.876.876 0 0 0-.4-.1c-.19 0-.42.13-.71.4-.28.27-1.17 1.49-1.43 1.76s-.48.4-.65.4c-.08 0-.19-.02-.32-.07s-1.45-.73-4.2-3.15-3.11-4-3.13-4.44c0-.17.13-.39.4-.65.28-.25.57-.51.89-.74.32-.24.61-.5.88-.78s.4-.52.4-.71c0-.13-.03-.27-.1-.4C7.12 4.32 4.62.19 4.53.1c-.19-.18-.92-.1-1.29.1C.25 1.82 0 4 .05 4.86c.05.89.61 5.58 5.2 9.93 5.7 5.41 9.66 5.2 9.92 5.2.87 0 3.52-.48 4.65-3.19.16-.38.31-1.07.09-1.29z"
                fillRule="evenodd"
              />
            </svg>
             
            Customer Care
          </a>
        </li>
      </ul>
    </div>
    <div
      className="sc-htpNat gQtKFA"
      style={
        Object {
          "padding": "0px 20px 0 0",
        }
      }
    >
      <section
        className="sc-bwzfXH irrVft"
      >
        <div
          className="sc-htpNat clwjnM"
          style={
            Object {
              "padding": "0px 10px",
            }
          }
        >
          <div
            className="sc-dnqmqq guHaqO"
          >
            <div
              style={
                Object {
                  "opacity": 0,
                  "transform": "translateY(-20px)",
                  "transition": "all 500ms ease-in-out",
                }
              }
            >
              <h1>
                Hello World
              </h1>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      className="sc-htpNat kpABsJ"
      style={
        Object {
          "padding": "0px 20px",
        }
      }
    >
      <section
        className="sc-bwzfXH irrVft"
      >
        <div
          className="sc-htpNat clwjnM"
          style={
            Object {
              "padding": "0px 10px",
            }
          }
        >
          <div
            className="sc-dnqmqq guHaqO"
          >
            <div
              style={
                Object {
                  "opacity": 0,
                  "transform": "translateY(-20px)",
                  "transition": "all 500ms ease-in-out",
                }
              }
            >
              <h1>
                Hello World
              </h1>
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>
</div>
`;

exports[`should render the layout 1`] = `
<Layout
  subtitle="Welcome to Otis payment portal"
  title="My Dashboard"
>
  <h1>
    Hello World
  </h1>
</Layout>
`;
